### DISCUSSION TITLE
How can you register your own helper?

### DISCUSSION THREAD

**Discussion by cacharbe** [farling42/obsidian-import-json#32] [Q&A] [answered]
I am new to using both Obsidian and Handlebars, so I might be missing something, but is it possible to define our own handlebar helpers? Would you do it in the md file directly?  Would it go into a code block?  I've tried quite a few permutations and can't seem to be able to get it to work.

**Comment by farling42** [comment, marked as answer]
At the moment I haven't created an API which will allow you to register your own HB helpers.

**Reply by cacharbe** [reply]
While unfortunate (feature request?) at least I'm not going crazy. It looks like using a lookup table will solve the immediate problem that I have, but being able to write inline handlers (or even includes) would be pretty powerful

**Reply by farling42** [reply]
I'm considering allowing a note to be specified which would contain the extra helper functions.

**Reply by cacharbe** [reply]
Fantastic.  Let me know if you need a tester.

**Reply by cacharbe** [reply]
looking at similar patterns might be worth using the scripts dir and structure the functions as js includes

**Comment by farling42** [comment]
Version 0.26.0 contains an additional field where a JS file can be specified in which some additional handlebars helper functions can be defined. See the README file for further information.
